File: algorithms/python/factorialTrailingZeroes/factorialTrailingZeroes.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-

# Source: https://leetcode.com/problems/factorial-trailing-zeroes/
# Author: Miao Zhang
# Date:   2021-01-24

class Solution:
    def trailingZeroes(self, n: int) -> int:
        res = 0
        while n >= 5:
            n //= 5
            res += n
        return res

File: algorithms/cpp/factorialTrailingZeroes/factorialTrailingZeroes.cpp
// Source: https://leetcode.com/problems/factorial-trailing-zeroes/
// Author: Miao Zhang
// Date:   2021-01-24

class Solution {
public:
    int trailingZeroes(int n) {
        int res = 0;
        while (n >= 5) {
            n /= 5;
            res += n;
        }
        return res;
    }
};
